docs/src/architecture/08_concepts/hermes_runtime_engine/hre_structure.md

```diff
@@ -0,0 +1,163 @@
+# HRE structure
+
+[Hermes Runtime Extension (*HRE*)][*HRE*] - stands as logically separate module (like a library) of the [*Hermes engine*]
+and provides an additional functionality to the Hermes runtime, therefore to [*Hermes application*].
+[WIT] files represent a source of truth of the [*Hermes events*] and *HRE api* definitions for a specific [*HRE*],
+and describe a standardized communication interface between [*Hermes application*]
+and [*Hermes engine's*][*Hermes engine*] runtime itself.
+
+Each [*HRE*] implementation take place inside `hermes/bin/srs/runtime_extensions` directory,
+for both Hermes related and WASI specific.
+
+Here is an example of the [WIT] files for the [hermes-cron] [*HRE*]:
+
+`world.wit`
+
+```wit
+package hermes:cron;
+
+world all {
+    import api;
+    export event;
+}
+```
+
+`event.wit`
+
+```wit
+interface event {
+    use api.{cron-event-tag, cron-tagged};
+
+    on-cron: func(event: cron-tagged, last: bool) -> bool;
+}
+
+world cron-event {
+    export event;
+}
+```
+
+`api.wit`
+
+```wit
+interface api {    
```

Comment:
Same as above, include code file contents, not a copy/paste of it.
You can also include just a section of the file if you want to truncate it like you do here.